prow/plugins/trigger/trigger.go

```diff
@@ -260,3 +258,33 @@ func skipRequested(c Client, pr *github.PullRequest, skippedJobs []config.Presub
 	}
 	return errorutil.NewAggregate(errors...)
 }
+
+func getPresubmits(c Client, pr *github.PullRequest) (string, []Presubmit, error) {
+	inRepoConfigEnabled := c.Config.InRepoConfigConfiguration(org, repo).Enabled
+	// We do the status context handling here instead of in the  PluginAgent to avoid
+	// updating it once to Pending and then Success/Failed per plugin that accesses the
+	// Presubmit config
+	if inRepoConfigEnabled {
+		// Create GitHub status context "prow-config-parsing" with status "Pending"
+	}
+	baseSHA, presubmits, err := c.GetPresubmits(pr)
+	if err != nil {
+		if inRepoConfigEnabled {
+			// Update GitHub status context "prow-config-parsing" with status "Failed"
+		}
+		return "", nil, err
+	}
+
+	if inRepoConfigEnabled {
+		// * There is exactly one case where we need to retire a context: If it was removed
+		//   on the base branch but is failed on our PR. We handle it by:
+		//   * Iterating over all existing contexts that do not have a "Success" status
+		//   * Ignore "tide" and "prow-config-parsing" contexts
+		//   * Ignore contexts whose `target_url` does not start with c.Config.GetJobUrlPrefix()
+		//   * check if found context is in presubmits if not, retire it
```

Comment:
We don't want to retire optional presubmits, presubmits that are not always_run, or statuses posted from 3rd party services so this isn't sufficient. We also need to be able to retire status contexts that are expected, but don't exist.

I think we should reuse the `unsuccessfulContexts` logic in Tide using the in-cluster config (not including PR changes) to identify contexts that are blocking merge, including missing contexts. We can then cross reference that with the presubmits we found from the in repo config and retire anything that isn't in the presubmits.

Reply:
> I think we should reuse the unsuccessfulContexts logic in Tide using the in-cluster config (not including PR changes) to identify contexts that are blocking merge, including missing contexts. We can then cross reference that with the presubmits we found from the in repo config and retire anything that isn't in the presubmits.

Oh that is actually great, as it also prevents tide from merging in case the context creation doesn't happen for whatever reason :+1: